src/test/java/ezvcard/io/json/roundtrip/JCardRoundTripTest.java

```diff
@@ -0,0 +1,155 @@
+package ezvcard.io.json.roundtrip;
+
+import static org.junit.Assert.*;
+
+import java.io.*;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import ezvcard.VCard;
+import ezvcard.VCardVersion;
+import ezvcard.io.json.JCardReader;
+import ezvcard.io.json.JCardWriter;
+import ezvcard.io.text.*;
+import ezvcard.util.IOUtils;
+
+/*
+ Copyright (c) 2012-2016, Michael Angstadt
+ All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions are met: 
+
+ 1. Redistributions of source code must retain the above copyright notice, this
+ list of conditions and the following disclaimer. 
+ 2. Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation
+ and/or other materials provided with the distribution. 
+
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+ The views and conclusions contained in the software and documentation are those
+ of the authors and should not be interpreted as representing official policies, 
+ either expressed or implied, of the FreeBSD Project.
+ */
+
+/**
+ * @author Buddy Gorven
+ */
+public class JCardRoundTripTest {
+	private final class ExtensionFilter implements FilenameFilter {
+		private final String ext;
+
+		private ExtensionFilter(String extension) {
+			ext = extension;
+		}
+
+		public boolean accept(File dir, String name) {
+			return name.toLowerCase().endsWith(ext);
+		}
+	}
+
+	@Rule
+	public TemporaryFolder tempFolder = new TemporaryFolder();
+
+	@Test
+	public void convert_to_jcard() throws Throwable {
+		File input = new File(JCardRoundTripTest.class.getResource("").toURI());
+		File[] dir = input.listFiles(new ExtensionFilter(".vcf"));
+		for (File vcf : dir) {
+			String file = vcf.getName().toString();
+			if (file.startsWith("John_Doe_EVOLUTION") || file.startsWith("John_Doe_IPHONE")
```

Comment:
Can you add a short comment here explaining why this "if" statement is here?

Is it here because of the problem with escaped characters in extended properties not being unescaped?  I guess I will fix this issue once I merge your pull request.  Then, I can remove this if statement.